src/utils.rs

```diff
@@ -33,6 +33,26 @@ pub(crate) fn trim_end_vec<T: PartialEq>(vec: &mut Vec<T>, value: &T) {
     vec.truncate(last_idx(vec, value));
 }
 
+/// Returns the highest power of `n` that fits in `u64`.
+#[inline]
+pub(crate) const fn max_pow_u64(n: u64) -> u64 {
+    match n {
+        2 | 8 => 1 << 63,
+        10 => 10_000_000_000_000_000_000,
+        16 => 1 << 60,
+        _ => max_pow_u64_impl(n),
+    }
+}
+
+#[inline]
+const fn max_pow_u64_impl(n: u64) -> u64 {
+    let mut max = n;
```

Comment:
nit: can we add a TODO here to use `ilog` when upgrading MSRV to 1.67 or greater?